src/Symfony/Component/Serializer/Normalizer/ArrayDenormalizer.php

```diff
@@ -46,12 +49,16 @@ public function denormalize($data, $class, $format = null, array $context = arra
         $serializer = $this->serializer;
         $class = substr($class, 0, -2);
 
-        return array_map(
-            function ($data) use ($serializer, $class, $format, $context) {
-                return $serializer->denormalize($data, $class, $format, $context);
-            },
-            $data
-        );
+        $builtinType = isset($context['key_type']) ? $context['key_type']->getBuiltinType() : null;
+        foreach ($data as $key => $value) {
+            if (null !== $builtinType && !call_user_func('is_'.$builtinType, $key)) {
+                throw new UnexpectedValueException(sprintf('The type of the key "%s" must be "%s" ("%s" given).', $key, $builtinType, gettype($key)));
```

Comment:
Brackets are used for the similar exception here https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php#L270

WSYT of standardizing messages with the rest of the code base in another PR?